dll/win32/shell32/CDefView.cpp

```diff
@@ -2616,11 +2667,11 @@ HRESULT WINAPI CDefView::DestroyViewWindow()
 
     if (m_hUpdateStatusbarThread)
     {
-        HANDLE hOldThread = m_hUpdateStatusbarThread;
+        HANDLE hOldThread = NULL;
 
         // Assigning NULL to m_hUpdateStatusbarThread will terminate the target thread
-        m_hUpdateStatusbarThread = NULL;
-        ::WaitForSingleObject(hOldThread, INFINITE);
+        hOldThread = ::InterlockedExchangePointer(&m_hUpdateStatusbarThread, hOldThread);
```

Comment:
For portability you should use interlocked functions at all points where this is accessed, since you imply here that there could be race conditions

Reply:
```cxx
    CDefView* pView = static_cast<CDefView*>(args);
    pView->UpdateStatusbarWorker(pView->m_hUpdateStatusbarThread);
    pView->Release();
```

How can I fix it?